src/libpgmoneta/utils.c

```diff
@@ -3003,6 +3004,87 @@ pgmoneta_is_file_archive(char* file_path)
    return false;
 }
 
+struct token_bucket*
+pgmoneta_token_bucket_init(unsigned long burst, unsigned long max_rate, unsigned long every)
+{
+
+   struct token_bucket* tb;
+   tb = (struct token_bucket*)malloc(sizeof(struct token_bucket));
+   if (tb == NULL)
+   {
+      return NULL;
+   }
+
+   tb->burst = burst;
+   atomic_init(&tb->cur_tokens, burst);
```

Comment:
Why do we need atomic on tokens and timestamps?

Reply:
Just in case, this algorithm may be utilized for multi-threading purposes in some instances.